config/peerpods/podvm/libvirt-podvm-image-handler.sh

```diff
@@ -0,0 +1,280 @@
+#!/bin/bash
+# FILEPATH: libvirt-podvm-image-handler.sh
+
+# This script is used to create or delete libvirt qcow2 image for podvm
+# The basic assumption is that the required variables are set as environment variables in the pod
+# Typically the variables are read from configmaps and set as environment variables in the pod
+# The script will be called with one of the following options:
+# Create image (-c)
+# Delete image (-C)
+
+set -x
+# include common functions from lib.sh
+# shellcheck source=/dev/null
+# The directory is where libvirt-podvm-image-handler.sh is located
+source /scripts/lib.sh
+
+# Function to verify that the required variables are set
+
+function verify_vars() {
+    # Ensure CLOUD_PROVIDER is set to libvirt
+    [[ -z "${CLOUD_PROVIDER}" || "${CLOUD_PROVIDER}" != "libvirt" ]] && error_exit "CLOUD_PROVIDER is empty or not set to libvirt"
+
+    [[ -z "${ORG_ID}" ]] && error_exit "ORG_ID is not set"
+    [[ -z "${ACTIVATION_KEY}" ]] && error_exit "ACTIVATION_KEY is not set"
+    [[ -z "${BASE_OS_VERSION}" ]] && error_exit "BASE_OS_VERSION is not set"
+
+    [[ -z "${PODVM_DISTRO}" ]] && error_exit "PODVM_DISTRO is not set"
+
+    [[ -z "${CAA_SRC}" ]] && error_exit "CAA_SRC is empty"
+    [[ -z "${CAA_REF}" ]] && error_exit "CAA_REF is empty"
+
+    [[ -z "${REDHAT_OFFLINE_TOKEN}" ]] && error_exit "Redhat token is not set"
```

Comment:
REDHAT_OFFLINE_TOKEN is not there in `libvirt-podvm-image-cm`

Reply:
I assume this is sensitive value, wouldn't be better to keep in the  Secret?